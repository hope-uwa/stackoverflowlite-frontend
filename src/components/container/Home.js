import React from 'react';
import QuestionsDisplay from './QuestionsDisplay';

class Home extends React.Component {
  render() {
    return (
            <main>
                {/* <div className="question-post" id="questionBanner"> 
                    <div className="container">
                        <div className="row">
                            <div className="question-box pl2 pr2 mt2 mb2 pls1">
                                <h3>Give it a Title:</h3>
                                <input type="text" className="w100" name="title" id="">
                                <h3>Question:</h3>
                                <textarea name="" className="w100"  id="" cols="30" rows="10" style="overflow-y: scroll"></textarea>
                                <div className="align-right mt1 mb1">
                                    <button id="cancel-question" onclick="cancelQuestion()" className="btn btn-danger btn-lg btn-shadow">Cancel</button>
                                    <button className="btn btn-success btn-lg btn-shadow">Publish</button>
                                </div>
                            </div>

                        </div>

                    </div>
                </div> */}
                <div className="container">
                    <div className="row">
                        <div className="col9 colsm12  col-order2 ">
                                <div className="row bg-cream align-left pt1 pl2 pb1 pls1 bb-grey">
                                    <div className="col9">
                                            <h1 id="thread-title" className="">Top Questions</h1>
                                    </div>
                                    <div className="col3">
                                        <div className="row pt1 pb1 ">
                                                <button id="askQuestion" onClick="signupModalBtn()" className="btn btn-success btn-md btn-shadow">
                                                    Ask a Question
                                                </button>
                                        </div>
                                            
                                    </div>
                                    
                                </div>
                                <div id="recent" className=" allquestion row pl2 pls1">
                                    <QuestionsDisplay />      
                                
                                </div>
                                <div className="pagination">
                                    <a href="#">❮</a>
                                    <a href="#">❯</a>
                                </div>
                        </div>
                        <div className="col3 colsm12 sidebar col-order1 " >
                            <div className="row mt3 mb2 show-md-hide-sm">
                                
                            </div>
                            <div className="row welcome show-md-hide-sm">
                                
                                
                            </div>
                            <div className="row h-400 h-sm-auto" >
                                    
                                    <ul className="align-center profile__actions">
                                        <li className="profile__actions-questions"><i className="dblue-text fa fa-fire"></i>&nbsp;<a href="#">Top Questions</a></li>
                                        <li className="profile__actions-recent"><i className=" dblue-text fa fa-filter"></i>&nbsp;<a href="#">Lastest Questions</a></li>
                                        <li className="profile__actions-threads"><i className=" dblue-text fa fa-comments"></i>&nbsp;<a href="#">All Questions</a></li>
                                        <li className="profile__actions-questions"><i className="dblue-text fa fa-fire"></i>&nbsp;<a href="#">Most Answered Questions</a></li>
                                        <li className="profile__actions-recent"><i className=" dblue-text fa fa-filter"></i>&nbsp;<a href="#">Recently Asked Questions</a></li>
                                        <li className="profile__actions-threads"><i className=" dblue-text fa fa-comments"></i>&nbsp;<a href="#">All Questions</a></li>
                                    </ul>
            
                                </div>
                        </div>
                    </div>


                </div>
                
                
            </main>
    );
  }
}

export default Home;