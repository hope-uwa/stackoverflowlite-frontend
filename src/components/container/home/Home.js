/* eslint-disable react/prop-types */
/* eslint-disable max-len */
/* eslint-disable class-methods-use-this */
import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { loadAllQuestion } from '../../../actions/allQuestions/allQuestionAction';
import AllQuestion from '../../presentation/allquestions/AllQuestions';
import Banner from '../banner/Banner';

export class Home extends React.Component {
  componentDidMount =() => {
    this.props.loadQuestion();
  }

  componentWillReceiveProps(nextProps) {
    if (JSON.stringify(this.props.user) !== JSON.stringify(nextProps.user)) {
      this.props.loadQuestion();
    }
  }


  viewQuestion = (questionId) => {
    const {
      history
    } = this.props;
    history.push(`/questions/${questionId}`);
  }

  render() {
    const { allQuestion } = this.props;

    return (
      <div>
        <Banner title={'Lastest Questions'}/>
        <div className=" allquestion row pl2 pls1">
          {allQuestion.map((x, key) => <AllQuestion questions={x} key={key} view={this.viewQuestion} />)}

        </div>
        <div className="pagination">
          <a href="#">❮</a>
          <a href="#">❯</a>
        </div>
      </div>

    );
  }
}

Home.propTypes = {
  history: PropTypes.object,
  loadQuestion: PropTypes.func,
  allQuestions: PropTypes.array,
};

export const mapStateToProps = (state) => {
  return {
    allQuestion: state.allQuestionReducer.questions,
    user: state.loginReducer
  };
};

export const mapDispatchToProps = {
  loadQuestion: () => loadAllQuestion()
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);
