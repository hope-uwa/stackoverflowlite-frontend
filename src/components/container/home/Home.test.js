import React from 'react';

import { shallow } from 'enzyme';
import { Home, mapDispatchToProps, mapStateToProps } from './Home';

describe('landing page', () => {
  const getProps = isLoading => (
    {
      allQuestion: [
        {
          title: 'this is the title',
        }],
      loadQuestion: () => {},
    }
  );
  it('should show page is loading', () => {
    const props = getProps(true);
    const component = shallow(<Home {...props}/>);
    expect(component).toMatchSnapshot();
  });
});
