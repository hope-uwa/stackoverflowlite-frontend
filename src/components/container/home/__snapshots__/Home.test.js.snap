// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing page should show page is loading 1`] = `
<div>
  <Banner
    title="Lastest Questions"
  />
  <div
    className=" allquestion row pl2 pls1"
  >
    <AllQuestions
      key="0"
      questions={
        Object {
          "title": "this is the title",
        }
      }
      view={[Function]}
    />
  </div>
  <div
    className="pagination"
  >
    <a
      href="#"
    >
      ❮
    </a>
    <a
      href="#"
    >
      ❯
    </a>
  </div>
</div>
`;
